jubilant/_juju.py

```diff
@@ -478,52 +478,66 @@ def deploy(
         if isinstance(overlays, str):
             raise TypeError('overlays must be an iterable of str or pathlib.Path, not str')
 
-        args = ['deploy', str(charm)]
-        if app is not None:
-            args.append(app)
+        charm = str(charm)
+        temp_needed = charm.startswith(('.', '/')) and self._juju_is_snap
```

Comment:
From in person discussion: for vm charm that need a resource, the latter too could be a local file.

Reply:
Good idea -- now done!